rdflib/namespace.py

```diff
@@ -330,6 +330,21 @@ def compute_qname(self, uri, generate=True):
             namespace, name = split_uri(uri)
             namespace = URIRef(namespace)
             prefix = self.store.prefix(namespace)
+            # check to see if there are longer prefixes defined
+            temp_namespace = namespace.toPython()
+            output_prefix = prefix
+            output_namespace = namespace
+            output_name = name
+            for i in xrange(1, len(name) + 1):
```

Comment:
So my first round of performance testing on the 3 different implementations produced some weird results which I think are not representative of the spectrum of workloads. I serialized https://github.com/SciCrunch/NIF-Ontology/blob/master/ttl/NIF-Molecule.ttl 10 times using 3 different implementations and average the time spend calling compute_qname in namespace.py as measured by cProfile in python3.5, there were 246680 calls to compute_qname for each run.
current mainline rdflib implementation of compute_qname: 1.2658s
loop over uri implementation (current for this pull request) of compute_qname: 1.1765s
loop on namespaces implementation of compute_qname: 1.2025s

This was a quick and dirty test and I am going to set it up a more robust suite since I don't believe that the implementation in question here is actually the fastest for all workloads as the numbers above would seem to suggest.